include/RAJA/index/IndexSet.hpp

```diff
@@ -86,9 +86,17 @@ template <typename T0, typename... TREST>
 class TypedIndexSet<T0, TREST...> : public TypedIndexSet<TREST...>
 {
   using PARENT = TypedIndexSet<TREST...>;
-  const int T0_TypeId = sizeof...(TREST);
+  static const int T0_TypeId = sizeof...(TREST);
 
 public:
+  // Adopt the value type of the first segment type
+  using value_type = typename T0::value_type;
+
+  // Ensure that all value types in all segments are the same
```

Comment:
I would really like to uncomment this static_assert, but it breaks a number of examples.
Basically: all segments in a TypedIndexSet should have the same value_type, otherwise something is wrong.